hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -267,19 +274,32 @@ def is_annotated_type(thing):
     )
 
 
-def find_annotated_strategy(annotated_type):  # pragma: no cover
-    flattened_meta = []
+def find_annotated_strategy(annotated_type):
+    # Annotated[None, ...].__origin__ is `NoneType`, so `None`
+    # can safely be used as a default value:
+    origin = getattr(annotated_type, "__origin__", None)
+    metadata = getattr(annotated_type, "__metadata__", ())
 
-    all_args = (
-        *getattr(annotated_type, "__args__", ()),
-        *getattr(annotated_type, "__metadata__", ()),
-    )
-    for arg in all_args:
-        if is_annotated_type(arg):
-            flattened_meta.append(find_annotated_strategy(arg))
+    if origin is None or metadata == ():
+        return None
```

Comment:
If `metadata == ()`, doesn't that mean we should `return st.from_type(origin)`?

Reply:
This is catched in `from_typing_type`:
https://github.com/HypothesisWorks/hypothesis/blob/a35c0cc89fb21b132993cdf56c2715966e572a90/hypothesis-python/src/hypothesis/strategies/_internal/types.py#L348-L354

But actually this line is a bit defensive, both `__origin__` and `__metadata__` should exist, so I can remove this check and the `getattr` default values